src/types.rs

```diff
@@ -193,19 +199,25 @@ macro_rules! key_methods {
                     write!(f, "0x")?
                 }
 
-                self.0.iter().try_for_each(|b| write!(f, "{:02X}", &b))
+                match f.width() {
+                    Some(w) if w > 0 => self.0.chunks(Self::LEN / w).try_for_each(|c| {
```

Comment:
Since each byte is two hex characters in a string (00-FF), shouldn't we divide these chunks in half? I.e. Self::LEN / w / 2?

Reply:
That does make sense. It was originally thought to be the amount of bytes, but instead we should focus on the string/formatted perspective

Added here: https://github.com/FuelLabs/fuel-types/pull/24/commits/7fbf7970557e18626c9de2f60d3eda91fb64855f

Used `2 · _` there because its the inverse of the len

Reply:
The only problem there is if the user puts there invalid values. Example: `1`. In that case, we will print `2` chars instead of `1` (since its impossible to fully represent bytes with one char).

We can leave this as tech debt for the future

Reply:
https://github.com/FuelLabs/fuel-types/issues/25